### ISSUE TITLE
Add per-user services

### ISSUE THREAD

**Issue by fpletz** [NixOS/nixpkgs#26128]
Currently all systemd user services defined by `systemd.user.services` are activated for all users.

It would be cool if we could toggle user services on a per-user basis.

See improvements on user services in #26094 and per-user packages support from #25712.

**Comment by CMCDragonkai** [comment]
These user services would still be configured in the root level configuration of `/etc/nixos/configuration.nix` right? So this is more for the root user to pre-configure user-level services rather than for the non-root user to declaratively configure their services independently of the root.

**Comment by florianjacob** [comment]
@CMCDragonkai You're right, this is indeed about pre-configuring a user-level service for a *single* user through `/etc/nixos/configuration.nix`, something like `systemd.user."alice".services`, as the current system only allows to configure a user service either for *every* user or for no one.

Declarative configuration for non-root users is the domain of projects like https://github.com/rycee/home-manager .

**Comment by hyperfekt** [comment]
From my research this is not feasible to do with NixOS because per-user units are placed in that user's tree directory. A tool like home-manager can do this.
The way to go with only NixOS is to create the unit file via the `systemd.user` option (which enables it for all user daemons) and add an appropriate `ConditionUser` field to limit execution to the intended user

**Comment by florianjacob** [comment]
I wasn't aware of this! That sounds like the right solution to this problem, as according to documentation, for non-`ConditionUser`-users, the service will be silently skipped. :+1:

**Comment by bufferUnderrun** [comment]
up !

I'm trying to achieve quite the same : running a podman container owned by a simple user and start at boot.
I'm stuck as describe in this issue (2017).

So, any solution in 2022 ?

Thanks
